src/network/mod.rs

```diff
@@ -170,6 +170,14 @@ impl Network {
         self.keys.get_matching(name).map(|entry| &entry.value.1)
     }
 
+    /// Returns the latest known key for a section with `prefix`.
+    /// If this returns `None` that means the latest known key is the genesis key.
+    pub fn key_by_prefix(&self, prefix: &Prefix) -> Option<&bls::PublicKey> {
```

Comment:
we have `section_by_name` already, why not just use `section_by_name(prefix.name())` ?

Reply:
Hm, I'm not sure that's a good idea. So say we call it with prefix `01` but we already progressed and have `010` and `011` in the key map. Then `name()` of `01` is `01000000....` which would be matched by `010` but not `011` even though both are valid successors to `01`. So why would we prefer one over the other?